### ISSUE TITLE
Pytorch dataloader bug

### ISSUE THREAD

**Issue by sidml** [jfpuget/STFT_Transformer#2]
I see that in the [dataset class](https://github.com/jfpuget/STFT_Transformer/blob/b462dd23de9d30cd44e3a71cc3349655dfdd4400/stft_transformer_final.py#L218) you are using numpy random number generator. Also, in the dataloader you aren't using worker_init_fn as recommend in [pytorch docs](https://pytorch.org/docs/stable/notes/randomness.html). This makes me suspect that the random samples are not being generated according to our expectations. More details about the bug are [here](https://pytorch.org/docs/stable/notes/randomness.html)

It will be great to see the impact of this bug on model performance.

**Comment by jfpuget** [comment]
With pytorch 1.9 there is no longer a need to initialize numpy random seed in data loaders: https://github.com/pytorch/pytorch/releases/tag/v1.9.0

This said I tried both ways with pytorch 1.8 that I was using for the competition.  It so happen that I got better results the way the code is in the repo.  It surprised me, but that's the way it is.

**Comment by sidml** [comment]
Yes, it seems to be fixed in Pytorch 1.9 :smile: 
I have also observed that using the "correct" way leads to slightly worse results. I wonder why :thinking:
